app/api/domains/osu.py

```diff
@@ -788,168 +788,179 @@ async def osuSubmitModularSelector(
         if not score.player.restricted:
             app.state.sessions.players.enqueue(app.packets.user_stats(score.player))
 
-    # stop here if this is a duplicate score
-    if await app.state.services.database.fetch_one(
-        "SELECT 1 FROM scores WHERE online_checksum = :checksum",
-        {"checksum": score.client_checksum},
-    ):
-        log(f"{score.player} submitted a duplicate score.", Ansi.LYELLOW)
-        return Response(b"error: no")
-
-    # all data read from submission.
-    # now we can calculate things based on our data.
-    score.acc = score.calculate_accuracy()
+    # hold a lock around (check if submitted, submission) to ensure no duplicates
```

Comment:
protip to make prs like this readable

![image](https://i.cmyui.xyz/ThwvUsT-9P31qit1bWJJ.png)